test/e2e/storage/regional_pd.go

```diff
@@ -0,0 +1,438 @@
+			testZonalFailover(c, ns)
+		})
+
+	})
+})
+
+func testVolumeProvisioning(c clientset.Interface, ns string) {
+	cloudZones := getRandomCloudZones(c)
+
+	// This test checks that dynamic provisioning can provision a volume
+	// that can be used to persist data among pods.
+	tests := []storageClassTest{
+		{
+			name:           "HDD Regional PD on GCE/GKE",
+			cloudProviders: []string{"gce", "gke"},
+			provisioner:    "kubernetes.io/gce-pd",
+			parameters: map[string]string{
+				"type":             "pd-standard",
+				"zones":            strings.Join(cloudZones, ","),
+				"replication-type": "repd",
+			},
+			claimSize:    "1.5G",
+			expectedSize: "2G",
+			pvCheck: func(volume *v1.PersistentVolume) error {
+				err := checkGCEPD(volume, "pd-standard")
+				if err != nil {
+					return err
+				}
+				return verifyZonesInPV(volume, cloudZones, true /* match */)
+			},
+		},
+		{
+			name:           "HDD Regional PD with auto zone selection on GCE/GKE",
+			cloudProviders: []string{"gce", "gke"},
+			provisioner:    "kubernetes.io/gce-pd",
+			parameters: map[string]string{
+				"type":             "pd-standard",
+				"replication-type": "repd",
+			},
+			claimSize:    "1.5G",
+			expectedSize: "2G",
+			pvCheck: func(volume *v1.PersistentVolume) error {
+				err := checkGCEPD(volume, "pd-standard")
+				if err != nil {
+					return err
+				}
+				return verifyZonesInPV(volume, framework.GetClusterZones(c).List(), false /* match */)
+			},
+		},
+	}
+
+	for _, test := range tests {
+		class := newStorageClass(test, ns, "" /* suffix */)
+		claim := newClaim(test, ns, "" /* suffix */)
+		claim.Spec.StorageClassName = &class.Name
+		testDynamicProvisioning(test, c, claim, class)
+	}
+}
+
+func testZonalFailover(c clientset.Interface, ns string) {
+	nodes := framework.GetReadySchedulableNodesOrDie(c)
+	nodeCount := len(nodes.Items)
+
+	cloudZones := getRandomCloudZones(c)
+	class := newRegionalStorageClass(ns, cloudZones)
+	claimTemplate := newClaimTemplate(ns)
+	claimTemplate.Spec.StorageClassName = &class.Name
+	statefulSet, service, repdLabels := newStatefulSet(claimTemplate, ns)
+
+	By("creating a StorageClass " + class.Name)
+	_, err := c.StorageV1().StorageClasses().Create(class)
+	Expect(err).NotTo(HaveOccurred())
+	defer func() {
+		framework.Logf("deleting storage class %s", class.Name)
+		framework.ExpectNoError(c.StorageV1().StorageClasses().Delete(class.Name, nil),
+			"Error deleting StorageClass %s", class.Name)
+	}()
+
+	By("creating a StatefulSet")
+	_, err = c.CoreV1().Services(ns).Create(service)
+	Expect(err).NotTo(HaveOccurred())
+	_, err = c.AppsV1().StatefulSets(ns).Create(statefulSet)
+	Expect(err).NotTo(HaveOccurred())
+
+	err = framework.WaitForStatefulSetReplicasReady(statefulSet.Name, ns, c, framework.Poll, statefulSetReadyTimeout)
+	if err != nil {
+		pod := getPod(c, ns, repdLabels)
+		Expect(podutil.IsPodReadyConditionTrue(pod.Status)).To(BeTrue(),
```

Comment:
Is this for more debugging?  Could you just Expect(err).NotTo(HaveOccurred())?

Reply:
Yeah. The error just indicates there was a timeout waiting for statefulset to become ready. In the case the test fails due to an actual bug, it's helpful to print out the pod state.